sys/trickle/trickle.c

```diff
@@ -33,25 +35,21 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t old_interval = trickle->I;
-    uint32_t max_interval = trickle->Imin << trickle->Imax;
-    uint32_t diff = old_interval - trickle->t;
+    assert(trickle->I > 0);
 
-    trickle->I *= 2;
+    uint32_t diff = trickle->I - trickle->t;
 
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
-        trickle->I = max_interval;
-    }
+    trickle->I *= 2;
+    trickle->I = MIN(trickle->I, (trickle->Imin << trickle->Imax));
 
     DEBUG("trickle: I == %" PRIu32 ", diff == %" PRIu32 "\n", trickle->I, diff);
 
     trickle->c = 0;
-    trickle->t = random_uint32_range(old_interval, trickle->I);
+    trickle->t = random_uint32_range(trickle->I/2, trickle->I);
```

Comment:
I don't see the benefit of **always** dividing `trickle->I` by two, whereas in the current implementation the old value (before multipying by 2) is stored in `old_value`. The only case, where a division by 2 is necessary, is at the corner case (I=Imax or I=0) and this is fixed by #7163.